mantidproject/mantid
# Figaro: loading cycle231 data fix

**Description of work.**

This PR implements the fix enabling loading of cycle 231 data generated by Figaro instrument at ILL. The issue stemmed by deprecating a NeXus field containing mid-chopper to sample distance, necessary to calculate TOF. The logic now checks if the new field, named `Distance.MidChopper_Sample` exists and loads it if it does, otherwise tries to load the `ChopperSetting.chopperpair_sample_distance` previously used to communicate this information. If neither are there, a runtime error exception is thrown.

**To test:**

1. The new unit test should be passing.
2. Try to load the file from the related issue.
3. There should be no issues.

Fixes #35309.

Release note added.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
